### PR TITLE
metrics: add `target_addr` labels to HTTP metrics

### PR DESCRIPTION
**Pull request by hawkw**
This branch builds on #861 and adds `target_addr` labels to HTTP metrics
as well as transport metrics. These labels contain a string
representation of the original destination address of the connection on
which that request was sent.

Additionally, I've made some more improvements to the metrics tests:
after the test code failed to catch a missing commma in labels, I
changed the metrics matching code to check that the metric is correctly
formatted, as well as containing the expected labels and values.
Furthermore, after rebasing onto `main` made some labels in tests added
in this PR incorrect, I modified the tests so that a majority of the
expected labels are defined in a single place by the text fuxtures,
since they don't actually differ between tests.

### CODE REVIEW

File: linkerd/app/core/src/metrics.rs [hunk 1]
```diff
@@ -266,6 +269,8 @@ impl FmtLabels for InboundEndpointLabels {
             write!(f, ",")?;
         }
 
+        write!(f, "target_addr=\"{}\"", self.target_addr)?;
```
Comment (olix0r) [inline linkerd/app/core/src/metrics.rs:272]: I would think this needs a trailing comma? Surprised our tests pass...

File: linkerd/app/core/src/metrics.rs [hunk 2]
```diff
@@ -279,6 +284,8 @@ impl FmtLabels for OutboundEndpointLabels {
...
```
Comment (olix0r) [inline linkerd/app/core/src/metrics.rs:287]: Similarly...

**Review by olix0r — CHANGES_REQUESTED** [review]

File: linkerd/app/core/src/metrics.rs (hunk 1 above)
Comment (hawkw) [reply linkerd/app/core/src/metrics.rs:272]: Yeah, commit c75b2d5d835615fd77f8a978c2fd5113af1fc3e4 updates the metrics test code to actually check that metric labels are formatted correctly --- the tests now fail with errors like
```
thread 'tests::telemetry::metrics_endpoint_inbound_request_count' panicked at 'malformed metric (multiple '=' in label)
 in label: "target_addr=\"127.0.0.1:34111\"tls=\"disabled\""
   metric: "request_total{direction="inbound",authority="tele.test.svc.cluster.local",target_addr="127.0.0.1:34111"tls="disabled"} 1"
  retried 5 times (73.354973ms)', linkerd/app/integration/src/metrics.rs:194:51
```

Good catch!

File: linkerd/app/integration/src/tests/telemetry.rs [hunk 3]
```diff
@@ -143,31 +143,65 @@ impl TcpFixture {
...
+}
+
+#[tokio::test]
+async fn metrics_endpoint_outbound_response_count() {
+    test_http_count("response_total", Fixture::outbound(), |proxy| {
+        let orig_dst = proxy.outbound_server.as_ref().unwrap().addr;
+        metrics::labels()
+            .label("direction", "outbound")
+            .label("tls", "no_identity")
+            .label("no_tls_reason", "not_provided_by_service_discovery")
```
Comment (olix0r) [inline linkerd/app/integration/src/tests/telemetry.rs:193]:
```suggestion
            .label("tls", "disabled")
```

**Review by olix0r — APPROVED** [review]

**Pull request merged**